kyaml/resid/gvk.go

```diff
@@ -57,27 +57,27 @@ func GvkFromString(s string) Gvk {
 			Kind:    noKind,
 		}
```

Comment:
The logic above
```go
if len(values) != 3 {
		// ...then the string didn't come from Gvk.String().
```		
doesn't hold anymore, since groups very commonly include periods. I think we'll need to get a bit fancier about this, and the GVK suite should have a few more realistic examples.

I confirmed experimentally that none of GVKNN can contain slashes. Both group and name can contain periods, but version, kind and namespace cannot.

Reply:
Thanks for the catch and for checking! I added some tests to gvk_test.go and resid_test.go to include groups and names with periods and changed the logic to account for these cases.